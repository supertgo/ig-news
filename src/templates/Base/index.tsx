import Header from 'components/Header';

import * as S from './styles';

export type BaseProps = {
  children: React.ReactNode;
};

const Base = ({ children }: BaseProps) => (
  <S.Wrapper>
    <Header />
    {children}
  </S.Wrapper>
);

export default Base;
