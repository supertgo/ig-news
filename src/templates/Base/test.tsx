import { render, screen } from '@testing-library/react';

import Base from '.';

jest.mock('components/Header', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Header"></div>;
    }
  };
});

describe('<Base />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Base>
        <h1>Base</h1>
      </Base>
    );

    expect(screen.getByRole('heading', { name: /Base/i })).toBeInTheDocument();
    expect(screen.getByTestId('Mock Header')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
