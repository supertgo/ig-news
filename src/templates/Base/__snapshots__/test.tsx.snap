// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render the heading 1`] = `
<div
  class=""
>
  <div
    data-testid="Mock Header"
  />
  <h1>
    Base
  </h1>
</div>
`;
